src/background/services/tabEvents.ts

```diff
@@ -96,18 +97,23 @@ export class TabEvents {
 
   onCreatedTab: CallbackTab<'onCreated'> = async (tab) => {
     if (!tab.id) return;
-    await this.updateVisualIndicators(tab.id);
+    await this.updateVisualIndicators(tab.id, tab.url);
   };
 
   updateVisualIndicators = async (
     tabId: TabId,
+    tabUrl?: string,
     isTabMonetized: boolean = tabId
       ? this.tabState.isTabMonetized(tabId)
       : false,
     hasTabAllSessionsInvalid: boolean = tabId
       ? this.tabState.tabHasAllSessionsInvalid(tabId)
       : false,
   ) => {
+    tabUrl ||= (await this.browser.tabs.get(tabId)).url;
```

Comment:
Let's not mutate the function parameter.

Reply:
How do you propose we set a value if not provided in this case? We can't use await in param defaults.

(I'm fine with mutating params, as long as we're not doing something like `param.foo = bar`, I mean, as long as callsite isn't affected)

Reply:
Hmm, do we even need the call to the `tabs` API? On newly created tabs the URL will default to a specific blank page `about:newtab`. What will be the case when the tab URL will not be defined in this case?